src/manager.js

```diff
@@ -307,50 +320,99 @@ AdManager.prototype.setSlotTargeting = function(element, slot) {
+        retval = '';
+
+      if (results) {
+        return decodeURIComponent(results[1].replace(/\+/g, ' '));
+      }
+
+      return retval;
+    },
+    expVar = (getExperimentVariation() !== null && getExperimentId() !== null) ? getExperimentId() + '_' + getExperimentVariation() : null,
+    paramZone = getQueryParameter('adzone'),
+    forcedAdZone = (function(pZone, meta) {
+      var tags = meta.tags,
+        forceNonCollapse = /why your team sucks|wyts/.test(tags),
+        forceCollapse = /nsfw|gamergate/.test(tags),
+        forceCollapseZone = !forceNonCollapse && forceCollapse ? 'collapse' : null,
+        post = meta.post,
+        postZone = post ? post.adZone : null;
+      return pZone || forceCollapseZone || postZone;
+      // kinjaMeta.post is not available at this time so we must reference kinja.postMeta
+    }(paramZone, postMeta)),
+
+    // figure out the defined adZone
+    adZone = forcedAdZone && forcedAdZone === 'collapse' ? 'collapse' : (pageType === 'frontpage' ? 'front' : pageType),
+    networkId = blogSales.adNetworkId || 4246,
+    unitName = '/' + [networkId, blogSales.adSiteName, adZone].join('/'),
+    adUnitConfig = this.adUnits.units[element.dataset.adUnit],
+    size = adUnitConfig.sizes[0][1],
+    tags = kinjaMeta.tags,
+    post = postMeta.post;
+
+  element.id = this.generateId();
+  slot = this.googletag.defineSlot(unitName, size, element.id);
+
   if (element.id && element.id in this.slots) {
     // Slot has already been configured
     return this.slots[element.id];
   }
 
-  element.id = this.generateId();
-
   if (!element.dataset) {
     this.logMessage('Browser does not support dataset', ERROR);
     return;
   }
 
-  var adUnitConfig = this.adUnits.units[element.dataset.adUnit];
   if (!adUnitConfig) {
     this.logMessage('Ad unit (' + element.dataset.adUnit + ') missing configuration', ERROR);
     return;
   }
 
-  var positionTargeting = adUnitConfig.slotName || element.dataset.adUnit;
-
-  var slotName = this.options.dfpSiteCode;
-  if (window.dfpSiteSection) {
-    slotName += '/' + window.dfpSiteSection;
-  }
-  var adUnitPath = '/' + this.options.dfpId + '/' + slotName;
-
-  // Use first ad size as the default
-  var size = adUnitConfig.sizes[0][1];
-  var slot = this.googletag.defineSlot(adUnitPath, size, element.id);
   slot.defineSizeMapping(adUnitConfig.sizes);
 
   if (slot === null) {
     // This probably means that the slot has already been filled.
     return;
   }
-
-  var slotTargeting = {};
-  if (element.dataset.targeting) {
-    slotTargeting = JSON.parse(element.dataset.targeting);
+  slotTargeting = targeting.getTargetingPairs(forcedAdZone);
+  
+  if (element.dataset.targeting && slotTargeting.slotOptions) {
+   utils.extend(slotTargeting.slotOptions, JSON.parse(element.dataset.targeting));
+  } else {
+	slotTargeting = JSON.parse(element.dataset.targeting);
+    slotTargeting.pos = positionTargeting;
+    element.dataset.targeting = JSON.stringify(slotTargeting);
```

Comment:
This is a bit tricky but i think this bit will allow bulbs to work, while still enabling kinja to use kvs set in the json targeting attr.